### PR TITLE
Flowchart & pseudocode

### PR DESCRIPTION
**Pull request by lstent**

### CODE REVIEW

**Comment by lstent** [comment]
Worksheet_C

File: worksheet_B/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Display start screen
+When player presses power initiate game
+Set correct word, add 11 incorrect words and 5> 11< symbol chains
+Symbol chain is radomly set to dud or extra attempts
+Set Player attempts left to 4
+
+Loop until winning screen or player attempts = 0
+  When player selects word
+    If word is correct go to winning screen
+    Else -1 attempts
```
Comment (edpowley) [inline worksheet_B/README.md:11]: The worksheet asked you to provide pseudocode for the calculation of the similarity score, i.e. loop through the words and count how many letters match between the guessed word and the secret word

File: worksheet_B/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  When player selects symbol chain
+    If chain is dud delete the chain and an uncorrect word if word hasn't been selected already
+    Else chain is deleted and attempts reset to 4
+When player attempts = 0 display looser screen
+```
+![flowchart] (Fallout Terminal Hacking Flowchart.pdf)
```
Comment (edpowley) [inline worksheet_B/README.md:17]: Flowchart looks good 👍

File: worksheet_C/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- (a) The algorithm checks list i and j for duplicates, if a duplicate is found it shows the word true, if not show false
```
Comment (edpowley) [inline worksheet_C/README.md:1]: i and j are not lists, they are indexes into a list

File: worksheet_C/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+- (b) For each item in list i it is being compared to every item before and including that same point in list j, e.g. list item 2 in i is being compared to list item 0,1,2 in j, written as n*n or n^2
```
Comment (edpowley) [inline worksheet_C/README.md:2]: ✅

File: worksheet_C/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+- (c) If line 4 is changed it still compares every item in list i and j for duplicates
```
Comment (edpowley) [inline worksheet_C/README.md:3]: Unclear -- needs more detail

File: worksheet_C/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+- (d) The algrorithm runs twice as fast because instead of comparing one item in list i against every item of list j, it only compares list numbers j below the list number i e.g. if the list number in i is 2 it only compares list number in j before 2
```
Comment (edpowley) [inline worksheet_C/README.md:4]: ✅

File: worksheet_C/README.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+- (e) I think the algorithm cannot be counted as quadratic anymore as the numbers are no longer being copared to its equivalent in list j therefore the equation is no longer n^2
```
Comment (edpowley) [inline worksheet_C/README.md:5]: As explained in class, the algorithm is still quadratic -- the number of comparisons is n(n-1)/2 which simplifies to O(n^2).

File: worksheet_C/README.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+- (f) Python's built-in source fuction is logarithmic written as O(n log n) ref: (2015) Timecomplexity [online] Available at: https://wiki.python.org/moin/TimeComplexity [Accessed 21st October 2016]
```
Comment (edpowley) [inline worksheet_C/README.md:6]: O(n log n) is correct, but this is not logarithmic -- logarithmic is O(log n), but this has an extra linear factor of n.

File: worksheet_C/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+- (g) The algorithm is working in a linear complexity as it just compares the list item above to the current list item, therefor in big O notation it is written as O(n)
```
Comment (edpowley) [inline worksheet_C/README.md:7]: However the algorithm also has to sort the list, so the time complexity is O(n log n + n) = O(n log n)

File: worksheet_C/README.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+- (h) In my opinion the linear algorithm would be faster as it quickly sorts itself and then only has to check the before list item for duplicates instead of going back over the whole of the previous items to check for duplicates like the quadratic algorithm does
```
Comment (edpowley) [inline worksheet_C/README.md:8]: ✅ (no need to say "in my opinion" -- it's a fact!)

File: worksheet_C/README.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+- (i) A programmer may choose a slower algorithm over a faster one as many faster algorithms depend of the use of sorting, therefore the list may not be already sorted which may take extra time to sort or the programmer may not know how to sort therefore it would just be faster to use a slower algorithm
```
Comment (edpowley) [inline worksheet_C/README.md:9]: Even factoring in the time it takes to sort, the second algorithm will still be faster than the first for large lists. "The programmer may not know how to sort" is an OK point, but a better one would be e.g. the "slower" algorithm is actually faster for small lists.

**Pull request closed without merging**